### ISSUE TITLE
Cache issue/timing issue with CompatibilityAccessManager when calling configure immediately after configuring the normal Purchases SDK.

### ISSUE THREAD

**Issue by valitovaza** [codykerns/PurchasesHelper#11]
In sandbox mode, everything works as expected, but in production, it seems that ‘CompatibilityAccessManager’ cannot convert old users without reopening the app. So my guess is that 

‘func isActive(entitlement: String, result: @escaping ((Bool, Purchases.PurchaserInfo?) -> Void))’
works only after some delay or 2nd time in production.

I configure entitlements on ‘appDidFinishLaunch’ to grand access with this method:

‘func configure(entitlements: [BackwardsCompatibilityEntitlement], completion: ((Purchases.PurchaserInfo?) -> Void)? = nil)’.
Also, I set ‘PurchasesDelegate’ to catch updates and configure UI.

**Comment by codykerns** [comment]
Can you confirm you're only checking for entitlements via CompatibilityAccessManager:
```swift
CompatibilityAccessManager.shared.isActive(entitlement: "entitlement") { (isActive, purchaserInfo) in
  // isActive is true if the user has access via RevenueCat or with registered build versions
}
```

And *not* via the default Purchases SDK property?
```swift
PurchaserInfo.entitlements["entitlement"].isActive
```

CompatibilityAccessManager is a local, client side check and doesn't change anything on RevenueCat's side, so you can't use the default properties on PurchaserInfo. In your case, it's likely only working after the cache expires or after the app is relaunched. In which case, you might have the versions registered incorrectly.

If indeed this is a timing issue, a workaround would be to manually call `Purchases.shared.syncPurchases` or `.restoreTransactions` at another time in your app's lifecycle, when we know that Purchases has been configured and PurchaserInfo has already been returned. 

You can see the exact logic for when CompatibilityAccessManager tries to sync a missing receipt here, that you can replicate in your own flow: https://github.com/codykerns/PurchasesHelper/blob/master/Sources/PurchasesHelper/CompatibilityAccessManager.swift#L50

Since CompatibilityAccessManager requires a receipt to be synced with RevenueCat in order to work (since we rely on the originalApplicationVersion in PurchaserInfo), we just need to try and get that user's receipt synced ASAP. If it's not working from the `configure` method, try and get purchases synced as soon as possible. 

Worst case scenario, you should communicate to your users in your paywall that if they are a previously paid user they need to restore purchases, and provide a button that calls `restoreTransactions`.

**Comment by codykerns** [comment]
I just published v0.2.0, which adds additional features including a new check for originalPurchaseDate, improves the ReadMe for clarity, as well as improves the logging of CompatibilityAccessManager. Take a look at the new version and see if things are a bit more clear for how to implement this package.

Closing this for now, feel free to re-open if you're still have issues.
